src/tests/nested-containers.test.ts

```diff
@@ -0,0 +1,650 @@
+import fetch from 'node-fetch';
+const { TextEncoder, TextDecoder } = require('util');
+
+// Code for gathering coverage for src/ not dist/, otherwise tests should test the built files in dist/
+let eosjs;
+let eosjs_jssig;
+if (process.env.COVERAGE_TESTING === 'true') {
+    eosjs = require('../');
+    eosjs_jssig = require('../eosjs-jssig');
+} else {
+    eosjs = require('../../dist');
+    eosjs_jssig = require('../../dist/eosjs-jssig');
+}
+
+const { JsonRpc, Api } = eosjs;
+const { JsSignatureProvider } = eosjs_jssig;
+
+const privateKey = '5JuH9fCXmU3xbj8nRmhPZaVrxxXrdPaRmZLW1cznNTmTQR2Kg5Z';
+
+const rpc = new JsonRpc('http://localhost:8888', { fetch });
+const signatureProvider = new JsSignatureProvider([privateKey]);
+const api = new Api({ rpc, signatureProvider, textDecoder: new TextDecoder(), textEncoder: new TextEncoder() });
+
+describe('eosjs-nested-container', () => {
+
+    it('should test Multi Index nested containers', async () => {
+        jest.setTimeout(30000);
+
+        await api.getAbi('nestcontnmi');
+
+        // Test action for set< set< uint16_t >>
+        await api.transact({
+            actions: [ api.with('nestcontnmi').as('alice').setstst('alice', [[10, 10], [3], [400, 500, 600]]) ]
+        }, { blocksBehind: 3, expireSeconds: 30 });
+
+        // Test action for set< vector< uint16_t >>
+        await api.transact({
+            actions: [ api.with('nestcontnmi').as('alice').setstv('alice', [[16, 26], [36], [36], [46, 506, 606]]) ]
+        }, { blocksBehind: 3, expireSeconds: 30 });
+
+        // Test action for set< optional< uint16_t >>
+        await api.transact({
+            actions: [ api.with('nestcontnmi').as('alice').setsto('alice', [null, null, 500]) ]
+        }, { blocksBehind: 3, expireSeconds: 30 });
+
+        // Test action for set< map< uint16_t, uint16_t >>
+        await api.transact({
+            actions: [ api.with('nestcontnmi').as('alice').setstm('alice', [[{'key':30,'value':300},{'key':30,'value':300}],[{'key':60,'value':600},{'key':60,'value':600}]]) ]
+        }, { blocksBehind: 3, expireSeconds: 30 });
+
+        // Test action for set< pair< uint16_t, uint16_t >>
+        await api.transact({
+            actions: [ api.with('nestcontnmi').as('alice').setstp('alice', [{'key': 69, 'value': 129}, {'key': 69, 'value': 129}]) ]
```

Comment:
The JSON input format here is NOT correct, for pair, 'first' and 'second' should be used, insstead of 'key' and 'value', Farhad is in the middle of updating the related .py script, see https://github.com/EOSIO/eos/pull/11006, so here all of the lines related to pair shall be changed.

Please note that the latest eosio-cpp built from [develop] branch of eosio.cdt shall be used, however, you can just run his *.py script, because he also modified/updated related .WASM and .abi generated by the latest eosio-cpp